configure

```diff
@@ -14,11 +15,16 @@ done
 set +C
 trap 'rm "$tmpc"' EXIT INT QUIT TERM HUP
 
+isgnuld() {
+	local ld="`$CC --print-prog-name=ld 2> /dev/null`" || return 1
```

Comment:
that approach is better, however i think we should do something like this
```
echo "int main() {return 0;}" > foo.c
$CC foo.c -Wl,--linker-option-that-solaris-ld-does-not-like || echo nope
```
```

Reply:
There are 2 options not recognized by Solaris ld `--no-as-needed` and `-soname=<name>`

`--no-as-needed` is not need for Solaris ld, since it is not have such bug to workaround by this option.

`-soname=<name>` could be replaced to either `-h <name>` or `--soname <name>`, and the later 2 forms are also accepted by GNU ld.

Reply:
ok, so let's try to add the above check and call it something like
```
whether linker supports -Wl,--no-as-needed ... yes/no
```
and then if its not supported we write `echo NO_AS_NEEDED= >>config.mak`

Reply:
in order to support this, we would need to modify `check_compile` function not to unconditionally put its argument into OUR_CFLAGS. that can go into a separate commit, since it's a logically separate change.